src/host_abi.zig

```diff
@@ -0,0 +1,108 @@
+/// All Roc functions that are exposed to the host take 1 argument and return void.
+/// The 1 argument is a pointer to one of these structs, which includes an address
+/// that the Roc call will write the return value into. This design makes Roc's ABI
+/// very simple; the calling convention is just "pass a single pointer".
+pub const RocCall = struct {
+    /// Function pointers that the Roc program uses, e.g. alloc, dealloc, etc.
+    ops: *RocOps,
+    /// The argument that the Roc entrypoint function will receive from the host.
+    /// (If multiple arguments are needed, this should be a pointer to a struct.)
+    arg: *anyopaque,
+    /// What the Roc entrypoint function will return to the host.
+    /// (The caller should have allocated enough space for the Roc function to write
+    /// the entire return value into this.)
+    ret: *anyopaque,
+};
+
+/// The operations (in the form of function pointers) that a running Roc program
+/// needs the host to provide.
+///
+/// This is used in both calls from actual hosts as well as evaluation of constants
+/// inside the Roc compiler itself.
+pub const RocOps = struct {
```

Comment:
Note, this also has to include every single effect function. I would suggest we just append those to the end in alphabetical order. So it will be bespoke per platform. 

...That or add a pointer to another struct containing all the effect pointers in alphabetical order.

Reply:
Good point! Added in 9245c1ea4257396d8185e85721f13ee03a824aec